Firestore/core/src/firebase/firestore/api/query_core.mm

```diff
@@ -294,12 +272,28 @@ QuerySnapshot result(firestore_, query_, std::move(snapshot),
     } else {
       // You can have at most 1 disjunctive filter and 1 array filter. Check if
       // the new filter conflicts with an existing one.
+      absl::optional<Operator> conflicting_op;
       Operator filter_op = field_filter.op();
-      bool is_array_op = absl::c_linear_search(kArrayOps, filter_op);
 
-      if (is_array_op && query_.HasArrayContainsFilter()) {
-        ThrowInvalidArgument("Invalid Query. Queries only support a single "
-                             "arrayContains filter.");
+      if (IsDisjunctiveOperator(filter_op)) {
+        conflicting_op = query_.FirstDisjunctiveOperator();
+      }
+      if (!conflicting_op.has_value() && IsArrayOperator(filter_op)) {
+        conflicting_op = query_.FirstArrayOperator();
+      }
+      if (conflicting_op.has_value()) {
```

Comment:
`absl::optional` behaves like a pointer. It can be evaluated in a boolean context and defines `operator*` so you can dereference it. This means that instead of writing this:

```c++
if (conflicting_op.has_value()) {
  if (conflicting_op.value() == filter_op) { ... }
}
```

You can write this like so:
```c++
if (conflicting_op) {
  if (*conflicting_op == filter_op) { ... }
}
```
Doing so isn't required, but is more idiomatic, and I thought I'd mention that you have the option.